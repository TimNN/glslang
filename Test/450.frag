#version 450 core

in float in1;
in vec2 in2;
in vec3 in3;
in vec4 in4;

void main()
{
    vec2 v2 = dFdxFine(in2);
    vec3 v3 = dFdyCoarse(in3);
    vec4 v4 = fwidth(in4);
    v4 = dFdyFine(in4);
    v3 = dFdyFine(in3);
    float f = dFdx(in1) + dFdxFine(in1) + dFdxCoarse(in1);
    v4 = fwidthCoarse(in4) + fwidthFine(in4);

    float cull = gl_CullDistance[2];
    float consts = gl_MaxCullDistances + gl_MaxCombinedClipAndCullDistances;
}
