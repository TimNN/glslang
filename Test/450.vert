#version 450 core

out gl_PerVertex {
    float gl_CullDistance[3];
};

void main()
{
    gl_CullDistance[2] = 4.5;
}
