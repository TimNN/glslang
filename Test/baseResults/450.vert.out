450.vert
Warning, version 450 is not yet complete; most version-specific features are present, but some are missing.

Shader version: 450
0:? Sequence
0:7  Function Definition: main( (void)
0:7    Function Parameters: 
0:9    Sequence
0:9      move second child to first child (float)
0:9        direct index (float)
0:9          gl_CullDistance: direct index for structure (out implicitly-sized array of float)
0:9            'anon@0' (out block{out implicitly-sized array of float gl_CullDistance})
0:9            Constant:
0:9              10 (const uint)
0:9          Constant:
0:9            2 (const int)
0:9        Constant:
0:9          4.500000
0:?   Linker Objects
0:?     'anon@0' (out block{out implicitly-sized array of float gl_CullDistance})
0:?     'gl_VertexID' (gl_VertexId int)
0:?     'gl_InstanceID' (gl_InstanceId int)


Linked vertex stage:


Shader version: 450
0:? Sequence
0:7  Function Definition: main( (void)
0:7    Function Parameters: 
0:9    Sequence
0:9      move second child to first child (float)
0:9        direct index (float)
0:9          gl_CullDistance: direct index for structure (out 3-element array of float)
0:9            'anon@0' (out block{out 3-element array of float gl_CullDistance})
0:9            Constant:
0:9              10 (const uint)
0:9          Constant:
0:9            2 (const int)
0:9        Constant:
0:9          4.500000
0:?   Linker Objects
0:?     'anon@0' (out block{out 3-element array of float gl_CullDistance})
0:?     'gl_VertexID' (gl_VertexId int)
0:?     'gl_InstanceID' (gl_InstanceId int)

